mappings/net/minecraft/client/gui/container/BeaconScreen.mapping

```diff
@@ -1,31 +1,60 @@
 CLASS daq net/minecraft/client/gui/container/BeaconScreen
-	CLASS daq$a WidgetButtonIconCancel
+	CLASS daq$a CancelButtonWidget
+		METHOD <init> (Ldaq;II)V
+			ARG 2 x
+			ARG 3 y
 		METHOD renderToolTip (II)V
 			ARG 1 mouseX
 			ARG 2 mouseY
-	CLASS daq$b WidgetButtonIconDone
+	CLASS daq$b DoneButtonWidget
+		METHOD <init> (Ldaq;II)V
+			ARG 2 x
+			ARG 3 y
 		METHOD renderToolTip (II)V
 			ARG 1 mouseX
 			ARG 2 mouseY
-	CLASS daq$c WidgetButtonIconEffect
+	CLASS daq$c EffectButtonWidget
 		FIELD b effect Laia;
+		FIELD c sprite Lduj;
 		FIELD d primary Z
+		METHOD <init> (Ldaq;IILaia;Z)V
+			ARG 2 x
+			ARG 3 y
+			ARG 4 statusEffect
+			ARG 5 primary
 		METHOD renderToolTip (II)V
 			ARG 1 mouseX
 			ARG 2 mouseY
-	CLASS daq$d WidgetButtonIcon
+	CLASS daq$d BaseButtonWidget
 		FIELD a disabled Z
+		METHOD <init> (II)V
+			ARG 1 x
+			ARG 2 y
+		METHOD a renderExtra ()V
 		METHOD a setDisabled (Z)V
+			ARG 1 disabled
 		METHOD b isDisabled ()Z
 		METHOD renderButton (IIF)V
 			ARG 1 mouseX
 			ARG 2 mouseY
 			ARG 3 delta
+	CLASS daq$e IconButtonWidget
+		FIELD a u I
+		FIELD b v I
+		METHOD <init> (IIII)V
+			ARG 1 x
+			ARG 2 y
+			ARG 3 u
+			ARG 4 v
 	FIELD k BG_TEX Lqs;
-	FIELD l doneButton Ldaq$b;
-	FIELD m canConsumeGem Z
+	FIELD l buttonDone Ldaq$b;
```

Comment:
Suffixing is preferred afaik, So why not `doneButton`?

Reply:
wait why did i do that, reverting the change